File: VB/RichEditDocumentServer_ImageSourceExample/CustomUriProvider.vb
#Region "#usings"
Imports DevExpress.Office.Services
Imports DevExpress.Office.Utils
Imports System
#End Region ' #usings

Namespace RichEditDocumentServer_ImageSourceExample
#Region "#customuriprovider"
    Public Class CustomUriProvider
        Implements IUriProvider

        Private Function IUriProvider_CreateImageUri(rootUri As String, image As OfficeImage, relativeUri As String) As String Implements IUriProvider.CreateImageUri
            Return image.Uri
        End Function

        Private Function IUriProvider_CreateCssUri(rootUri As String, styleText As String, relativeUri As String) As String Implements IUriProvider.CreateCssUri
            Return String.Empty
        End Function
    End Class
#End Region ' #customuriprovider    
End Namespace

File: CS/RichEditDocumentServer_ImageSourceExample/CustomUriProvider.cs
#region #usings
using DevExpress.Office.Services;
using DevExpress.Office.Utils;
using System;
#endregion #usings

namespace RichEditDocumentServer_ImageSourceExample
{
        #region #customuriprovider
        public class CustomUriProvider : IUriProvider
        {
            public string CreateCssUri(string rootUri, string styleText, string relativeUri)
            {
                return String.Empty;
            }

            public string CreateImageUri(string rootUri, OfficeImage image, string relativeUri)
            {
                return image.Uri;
            }
        }
        #endregion #customuriprovider    
}
